### PR TITLE
Add a forbid-dom-props rule

### PR DESCRIPTION
**Pull request by davazp**
A new rule forbid-dom-props was added. It is based on
forbid-component-props but it only forbid property on DOM Node
elements.

This can be useful to enforce some good practices, as not using 'id'
attributes or inline styles.

### CODE REVIEW

**Comment by davazp** [comment]
I used `dom` as they keyword for this rule as I didn't find any other rule acting only on DOM Nodes, and element refers to both Component and DOM Nodes.  Hopefully it is a good name!

I can imagine a more flexible set of option where you can forbid some properties based on the specific DOM Node, but I preferred to keep it similar to  `forbid-component-props` for now. I guess it can be extended if somebody finds that useful.

File: lib/rules/forbid-dom-props.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      type: 'object',
+      properties: {
+        forbid: {
+          type: 'array',
+          items: {
+            type: 'string'
+          }
+        }
+      },
+      additionalProperties: true
```
Comment (ljharb) [inline lib/rules/forbid-dom-props.js]: this should be set to `false`

File: lib/rules/forbid-dom-props.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      description: 'Forbid certain props on DOM Nodes',
+      category: 'Best Practices',
+      recommended: false
+    },
+
+    schema: [{
+      type: 'object',
+      properties: {
+        forbid: {
+          type: 'array',
```
Comment (ljharb) [inline lib/rules/forbid-dom-props.js]: this array should be forced to be unique

File: lib/rules/forbid-dom-props.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+          items: {
+            type: 'string'
```
Comment (ljharb) [inline lib/rules/forbid-dom-props.js]: these items should have a minLength of 1

**Comment by davazp** [comment]
@ljharb There were tests for the 3 cases already I think?  I have anyway added extra SFC and class-based  *invalid* tests, that there was just one.

All code was derived from `forbid-component-props`, including the schema.  Would you like me to change also the schema of `forbid-component-props` to follow your requested changes as well?  Or keep it for backward compatibility?

Thanks,
David.

**Comment by davazp** [comment]
I have also added Related section to the components to link to each other, following the example from `default-props-match-prop-types.md`.

**Comment by ljharb** [comment]
Changing the schema of the existing rule would be a breaking change; leave that the same, and let's update this one.

**Comment by davazp** [comment]
I updated the PR with the refined schema 👍

File: lib/rules/forbid-dom-props.js (hunk 3 above)
Comment (ljharb) [reply lib/rules/forbid-dom-props.js]: sorry, to clarify: 0 items in the array is OK, but an empty string should not be.

**Comment by davazp** [comment]
I misread your original comment. I fixed the schema now.

**Pull request merged**